src/libraries/System.Linq/src/System/Linq/Max.cs

```diff
@@ -37,6 +42,57 @@ public static int Max(this IEnumerable<int> source)
             return value;
         }
 
+        private static int Max(int[] array)
```

Comment:
So what I'd really like to see here is for many of these methods to be available on `Span<T>` and for us to forward this to the Span implementation.

Exposing new public APIs is of course something that needs to goto API proposal, but I think it would be better long term, would also allow more logic sharing for various helper APIs, and would centralize most of the "array like SIMD algorithms" to the same place in the BCL.

Reply:
I'd be fine subsequently seeing such methods on MemoryExtensions and deleting the code from here to delegate there instead.  Someone would need to do the work to figure out what that set of methods should be.  I also don't think it precludes adding this now and then delegating later if/when the methods exist.

Reply:
Right, I just wanted to call out what I'd like to see here and I think just about any method we think is worth special-casing for `array` in `LINQ` is applicable.

Sum/Min/Max are the "most obvious ones"